File: 000401-000600/000561.数组拆分I/561.go
package main

import (
	"fmt"
	"sort"
)

// 方法一：用时 56 ms
// 执行用时超过了 97 %
// 消耗内存超过了 25 %
func arrayPairSum(nums []int) int {
	sort.Ints(nums)
	ret := 0
	for i := 0; i < len(nums); i+=2 {
		ret += nums[i]
	}
	return ret
}

func main ()  {
	fmt.Println(arrayPairSum([]int{1, 4, 2, 3}))
	fmt.Println(arrayPairSum([]int{6, 2, 6, 5, 1, 2}))
}

File: 000401-000600/000561.数组拆分I/561.js
/**
 * @param {number[]} nums
 * @return {number}
 */
 // 方法一：用时 104 ms
 // 执行用时超过了 97 %
 // 消耗内存超过了 71 %
var arrayPairSum = function(nums) {
  nums = nums.sort((a, b) => (a - b))
  let ret = 0
  for (let i = 0; i < nums.length; i+=2) {
    ret += nums[i]
  }
  return ret
};

console.log(arrayPairSum([1, 4, 2, 3]))
console.log(arrayPairSum([6, 2, 6, 5, 1, 2]))
